### PR TITLE
VACMS-13851: Creates Bundle Classes incorporating functionality of EntityEventSubscriber

### LINKED ISSUE #13851: Shift complex content release entity event logic into bundle classes.
## Description

This is to help simplify the original code and the tests for `Drupal\va_gov_build_trigger\EventSubscriber\EntityEventSubscriber`. This class exists, the code's good, there are unit tests. 

But I want to refactor most of this code into a structure that can be tested more easily. If you look at the unit tests for this class, it will make your heart break. A lot of that is because we're performing so many operations directly on Node objects, which are brutally difficult to write mocks for.

Instead, we should shift the complex logic into [Bundle Classes](https://www.drupal.org/node/3191609), ensuring that we define meaningful interfaces covering this functionality, and then implementing those interfaces in the bundle classes.

For instance, for this method:

```php
 /**
   * Checks if a node is content type that is a facility.
   *
   * @param \Drupal\node\NodeInterface $node
   *   The node being altered.
   *
   * @return bool
   *   TRUE if it is a content type is a facility.  FALSE otherwise.
   */
  protected function isFacility(NodeInterface $node): bool {
    $facility_content_types = [
      'health_care_local_facility',
      // 'nca_facility',  // Not rendered on the FE yet.  Add it when it is.
      // 'vba_facility',  // Not rendered on the FE yet.  Add it when it is.
      'vet_center_cap',
      'vet_center_outstation',
      'vet_center',
    ];
    return in_array($node->getType(), $facility_content_types);
  }
```

We would do the following:

1. Define a `VaNodeInterface` interface, containing an `isFacility()` method.
2. Define a `VaNode` class, extending `Node` and implementing `VaNodeInterface` and `isFacility()` as including the logic above.

## Acceptance Criteria

- [x] We have evaluated the process of shifting complex logic from the above class into Bundle Classes.
- [x] If the process seems realistic and effective, either:
  - [x] the work has been done
  - [ ] followup ticket(s) opened for further work

### PR DESCRIPTION
**Pull request by ndouglas**
## Description

Closes #13851. 

A followup PR will provide a simplified service that should have total behavioral parity with the current `EventSubscriber`, once the code added to these bundle classes is factored in.

These changes do not as yet affect the content release system, so risk should be low.

## QA Steps

- [x] Automated tests pass.

### CODE REVIEW

File: docroot/modules/custom/va_gov_build_trigger/src/EventSubscriber/EntityEventSubscriber.php [hunk 1]
```diff
@@ -7,8 +7,8 @@
 use Drupal\core_event_dispatcher\Event\Entity\AbstractEntityEvent;
 use Drupal\node\NodeInterface;
 use Drupal\va_gov_build_trigger\Environment\EnvironmentDiscovery;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Drupal\va_gov_build_trigger\Service\BuildRequesterInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_build_trigger/src/EventSubscriber/EntityEventSubscriber.php:11]: Just alphabetizing.

File: config/sync/core.extension.yml [hunk 2]
```diff
@@ -237,6 +237,7 @@ module:
   va_gov_bulk: 0
   va_gov_clp: 0
   va_gov_content_release: 0
+  va_gov_content_types: 0
```
Comment (ndouglas) [inline config/sync/core.extension.yml:240]: A new module 🎉

**Comment by ndouglas** [comment]
Cypress failure was an unrelated 502 😩

**Review by edmund-dunn — APPROVED** [review]
Good work! Clean code!

**Pull request merged**